chalk-solve/src/clauses/super_traits.rs

```diff
@@ -0,0 +1,130 @@
+use rustc_hash::FxHashSet;
+
+use super::builder::ClauseBuilder;
+use crate::RustIrDatabase;
+use chalk_ir::{
+    fold::shift::Shift, interner::Interner, Binders, BoundVar, DebruijnIndex, TraitId, TraitRef,
+    WhereClause,
+};
+
+pub(super) fn push_where_clause_binders<I: Interner>(
+    builder: &mut ClauseBuilder<'_, I>,
+    binders: Binders<WhereClause<I>>,
+) {
+    builder.push_binders(binders, |builder, bound| match &bound {
+        // For the implemented traits, we need to elaborate super traits and add where clauses from the trait
+        WhereClause::Implemented(trait_ref) => {
+            push_trait_super_clauses(builder.db, builder, trait_ref.clone())
+        }
+        WhereClause::AliasEq(_) => builder.push_fact(bound),
```

Comment:
So, I'm of mixed minds here. The fact that there is the "Associated item bindings are just taken as facts (?)" comment where this bit of code came from tells me that this bit might be underspecified. Can you leave a FIXME instead in the two places when you re-inline? For a bit better visibility.